src/data_types/link_secret.rs

```diff
@@ -0,0 +1,56 @@
+use std::fmt;
+
+use crate::error::ConversionError;
+use serde::{Deserialize, Serialize};
+use ursa::cl::{prover::Prover as UrsaProver, MasterSecret};
+
+#[derive(Serialize, Deserialize)]
+pub struct LinkSecret {
+    pub value: MasterSecret,
```

Comment:
The structure is still:

```json
{
  "value": { ms: "BIGNUMBER" }
}
```

Reply:
Is it needed to keep this structure? I didn't see it referenced in the spec and I think it would be a good opportunity to simplify it to avoid confusions (IMO it should be simply a plain string)

Reply:
It's internal so I don't think it matters. Changing this should be doable with serde remote, but will be done if it's required.

Reply:
> I think it does matter, because when dealing with the library we usually call `load` or `toJson`, which in this case is misleading because we don't have any reference regarding how the expected format should be. In mi opinion these methods do make sense for standardized fields (i.e. the ones whose structure is clearly stated in AnonCreds spec).
> 
> As a direct user of this library, I can think of treating it as an opaque string and don't care about it's internal structure. However, if I want to use a Link Secret generated by other means or if I want to use one generated here for another purpose I wouldn't expect it to be a JSON object with such particular structure.
> 
> It feels more natural and logic to me if this `load` that receives a string would simply receive the Link Secret value instead of a JSON structure. Likewise, the `toJson` that outputs a string might become `toString`.

I agree with the structure feeling incorrect from a to/from json perspective. But `fromJson` taking a string as input feels very inconsistent with the API and every other implementation. 

I think reworking the structure to this:

```json
{
   "link_secret": "BIGNUMBER"
}
```

would be good. I think avoiding the `link_secret` in the JSON, by using a `fromString` method, would only cause confusing with regards to the rest of the API.

Reply:
Latest changes now include dealing with the link secret as a sting in decimal layout (like in the datamodels).